ToolBox/PsychSignalDataConverter/PsychSignalDataConverter.cs

```diff
@@ -0,0 +1,270 @@
+        /// <summary>
+        /// Filter raw files to be inclusive within the date range specified and converts those files.
+        /// </summary>
+        /// <param name="startDateUtc">Date to start parsing files from</param>
+        /// <param name="endDateUtc">Date to stop parsing files</param>
+        /// <remarks>Inclusive on the lower bound, and exclusive on the upper bound</remarks>
+        public void ConvertFrom(DateTime startDateUtc, DateTime endDateUtc)
+        {
+            // Filter for files by name and bounds, then order by date
+            var files = _rawSourceDirectory.GetFiles("*.csv", SearchOption.TopDirectoryOnly)
+                .Where(
+                    x =>
+                    {
+                        DateTime fileDate;
+                        if (!DateTime.TryParseExact(x.Name.Substring(0, 11), "yyyyMMdd_HH", CultureInfo.InvariantCulture, DateTimeStyles.AdjustToUniversal, out fileDate))
+                        {
+                            return false;
+                        }
+
+                        return fileDate >= startDateUtc && fileDate < endDateUtc;
+                    }
+                )
+                .OrderBy(x => DateTime.ParseExact(x.Name.Substring(0, 11), "yyyyMMdd_HH", CultureInfo.InvariantCulture, DateTimeStyles.AdjustToUniversal));
+
+            foreach (var rawFile in files)
+            {
+                Convert(rawFile);
+            }
+            
+            FlushAll();
+            CompressData();
+        }
+
+        /// <summary>
+        /// Iterate over the data, processing each data point before finally zipping the directories
+        /// </summary>
+        public void ConvertDirectory()
+        {
+            // Filter for raw data with file names formatted as "yyyyMMdd_HH.*"
+            // `GetDirectory()` doesn't guarantee file order, so we must order it manually ourselves
+            var files = _rawSourceDirectory.GetFiles("*.csv", SearchOption.TopDirectoryOnly)
+                .Where(
+                    x =>
+                    {
+                        DateTime fileDate;
+                        return DateTime.TryParseExact(x.Name.Substring(0, 11), "yyyyMMdd_HH", CultureInfo.InvariantCulture, DateTimeStyles.AdjustToUniversal, out fileDate);
+                    }
+                )
+                .OrderBy(x => DateTime.ParseExact(x.Name.Substring(0, 11), "yyyyMMdd_HH", CultureInfo.InvariantCulture, DateTimeStyles.AdjustToUniversal));
+
+            foreach (var rawFile in files)
+            {
+                Convert(rawFile);
+            }
+
+            FlushAll();
+            CompressData();   
+        }
+
+        /// <summary>
+        /// Flushes all open <see cref="StreamWriter"/> instances.
+        /// Should be called once the data has finished processing
+        /// </summary>
+        public void FlushAll()
+        {
+            foreach (var handle in _fileHandles.Values)
+            {
+                handle.Dispose();
```

Comment:
The method name `FlushAll()` and it's behavior _dispose_ all seems inconsistent. Also, after disposing all of the handles, we don't remove them  from the dictionary. If this is intended to simply be a flush operation, then lets change this to call a new method, `TickerData.Flush` and update `PsychSignalDataConverter` to implement `IDisposable` and that `Dispose` implementation can perform final disposal of the streams.